src/QsCompiler/DocumentationGenerator/Extensions.cs

```diff
@@ -0,0 +1,419 @@
+                                    ResolvedType.New(QsTypeKind<ResolvedType, UserDefinedType, QsTypeParameter, CallableInformation>.String)
+                                )
+                            )
+                        ),
+                        new InferredExpressionInformation(false, false),
+                        QsNullable<Range>.Null
+                    )
+                )
+            );
+
+        internal static string ToSyntax(this ResolvedType type) =>
+            SyntaxTreeToQsharp.Default.ToCode(type);
+
+        internal static string ToSyntax(this QsTypeItem item) =>
+            item switch
+            {
+                QsTypeItem.Anonymous anon => anon.Item.ToSyntax(),
+                QsTypeItem.Named named => $"{named.Item.VariableName.Value} : {named.Item.Type.ToSyntax()}"
+            };
+        
+        internal static string ToSyntax(this QsTuple<QsTypeItem> items) =>
+            items switch
+            {
+                QsTuple<QsTypeItem>.QsTuple tuple => $@"({
+                    String.Join(", ", tuple.Item.Select(innerItem => innerItem.ToSyntax()))
+                })",
+                QsTuple<QsTypeItem>.QsTupleItem item => item.Item.ToSyntax()
+            };
+
+        internal static string ToSyntax(this QsTuple<LocalVariableDeclaration<QsLocalSymbol>> items) =>
+            items switch
+            {
+                QsTuple<LocalVariableDeclaration<QsLocalSymbol>>.QsTuple tuple => $@"({
+                    String.Join(", ", tuple.Item.Select(innerItem => innerItem.ToSyntax()))
+                })",
+                QsTuple<LocalVariableDeclaration<QsLocalSymbol>>.QsTupleItem item => item.Item.ToSyntax()
+            };
+
+        internal static string ToSyntax(this LocalVariableDeclaration<QsLocalSymbol> symbol) =>
+            $@"{symbol.VariableName switch
+            {
+                QsLocalSymbol.ValidName name => name.Item.Value,
+                _ => "{{invalid}}"
+            }} : {symbol.Type.ToSyntax()}";
+
+        internal static string ToSyntax(this QsCustomType type) =>
+            $@"newtype {type.FullName.Name.Value} = {
+                String.Join(",", type.TypeItems.ToSyntax())
+            };";
+
+        internal static string ToSyntax(this ResolvedCharacteristics characteristics) =>
+            characteristics.SupportedFunctors.ValueOr(null) switch
+            {
+                null => "",
+                { Count: 0 } => "",
+                var functors => $@" is {String.Join(" + ", 
+                    functors.Select(functor => functor.Tag switch
+                    {
+                        QsFunctor.Tags.Adjoint => "Adj",
+                        QsFunctor.Tags.Controlled => "Ctl"
+                    })
+                )}"
+            };
+
+        internal static string ToSyntax(this QsCallable callable)
```

Comment:
Aren't all these ToSyntax methods for displaying the Q# representation as it would be in the source code? 
Please use SyntaxTreeToQsharp wherever possible instead.